packages/shared/src/hooks/useChecklist.ts

```diff
@@ -0,0 +1,51 @@
+import { useCallback, useEffect, useMemo, useState } from 'react';
+import { ChecklistStepType } from '../lib/checklist';
+import { Action, ActionType } from '../graphql/actions';
+
+export type UseChecklistProps = {
+  steps: Pick<ChecklistStepType, 'action'>[];
+};
+
+export type UseChecklist = {
+  openStep: string | undefined;
+  onToggleStep: (action: Action) => void;
+  isDone: boolean;
+  activeStep: ActionType | undefined;
+};
+
+const useChecklist = ({ steps }: UseChecklistProps): UseChecklist => {
+  const activeStep = useMemo(
```

Comment:
Not 100% sure, but maybe it's also something for the hook?
In my opinion all calculation regarding to steps should happen in that isolated scoped, but let me know what you think.  (pros/cons)

Reply:
Actually my main point here is that I wish to separate base checklist logic vs specific steps logic. 

In the future we will have other checklist for new features that we don't even know about and this is where that kind of separation will hopefully pay out. 

For example active step for the checklist will be the same always, it is the next uncompleted step in steps list and it is defined by spec/design. Checklist does not know how many more steps are there or what they do, it just knows it needs to show active/next step to the user. 

This is also why each `ChecklistStep` can accept `component` where you can customize what each step does on per case by case basis.

So to conclude `useChecklist` and `Checklist*` components contains all the generic checklist logic that is tied to checklist UI and then steps creation is something that will be done separately for each new checklist eg. for squads we have `useSquadChecklistSteps` and `SquadChecklistCard` where that specific logic lives.

Hope it makes more sense.

Reply:
Ah yes got it, so in that case we do need to ensure all calculations happen from the relevant hook, not the UI endpoints correct? (so useSquadChecklist)

Reply:
Yep, I streamlined everything even more in latest commit. Same as composing `Checklist*` components into `SquadChecklistCard` I did the same with `useChecklist` hook inside `useSquadChecklist`.